lib/features/push_notification/presentation/controller/push_base_controller.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:core/utils/app_logger.dart';
+import 'package:dartz/dartz.dart';
+import 'package:fcm/model/type_name.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart' as jmap;
+import 'package:jmap_dart_client/jmap/core/user_name.dart';
+import 'package:tmail_ui_user/features/base/action/ui_action.dart';
+import 'package:tmail_ui_user/features/push_notification/presentation/action/push_notification_state_change_action.dart';
+import 'package:tmail_ui_user/features/push_notification/presentation/listener/email_change_listener.dart';
+import 'package:tmail_ui_user/features/push_notification/presentation/listener/mailbox_change_listener.dart';
+
+abstract class PushBaseController {
+  Session? session;
+  AccountId? accountId;
+
+  StreamSubscription<Either<Failure, Success>>? _stateStreamSubscription;
+
+  void consumeState(Stream<Either<Failure, Success>> newStateStream) {
+    _stateStreamSubscription = newStateStream.listen(
+      _handleStateStream,
+      onError: handleErrorViewState,
+    );
+  }
+
+  void _handleStateStream(Either<Failure, Success> newState) {
+    newState.fold(handleFailureViewState, handleSuccessViewState);
+  }
+
+  void handleFailureViewState(Failure failure);
+
+  void handleSuccessViewState(Success success);
+
+  void handleErrorViewState(Object error, StackTrace stackTrace) {
+    logError('PushBaseController::handleErrorViewState():error: $error | stackTrace: $stackTrace');
+  }
+
+  void cancelStateStreamSubscription() {
+    _stateStreamSubscription?.cancel();
+    _stateStreamSubscription = null;
+  }
+
+  void initialize({AccountId? accountId, Session? session}) {
+    this.accountId = accountId;
+    this.session = session;
+  }
+
+  void mappingTypeStateToAction(
+    Map<String, dynamic> mapTypeState,
+    AccountId accountId,
+    UserName userName, {
+    bool isForeground = true,
```

Comment:
in web? Default value for it is true? Re-test this case in web: foreground tab, background tab, minimized browser

Reply:
<details>
<summary>Foreground tab</summary>

https://github.com/user-attachments/assets/a827000b-2f16-4264-9399-7168f36b6d08

</details>

<details>
<summary>Background tab</summary>

https://github.com/user-attachments/assets/1d93c8fc-78f2-4560-bbba-5a7fa559f2a1

</details>

<details>
<summary>Minimized</summary>

https://github.com/user-attachments/assets/8750ca8b-7ed4-434f-a0e1-8af9bfbb7317

</details>